docs/pages/admin-guides/infrastructure-as-code/terraform-provider/terraform-cloud.mdx

```diff
@@ -0,0 +1,296 @@
+---
+title: Run the Teleport Terraform Provider on Terraform Cloud
+description: How to manage dynamic resources using the Teleport Terraform provider on HCP Terraform, Terraform Cloud, and Terraform Enterprise.
+---
+
+This guide demonstrates how to use the Terraform provider for Teleport using
+HCP Terraform or Terraform Enterprise.
+
+This guide does not cover running the Terraform provider locally, in other CI/CD
+environments, or in short-lived cloud VMs. In any of these cases, refer to a
+dedicated guide:
+
+- [Run the Terraform Provider in CI or cloud VMs](./ci-or-cloud.mdx)
+- [Run the Terraform Provider locally](./local.mdx)
+
+## How it works
+
+<Admonition type="warning" title="Enterprise">
+  Terraform Cloud joining with self-hosted Terraform Enterprise requires
+  Teleport Enterprise. Terraform Cloud joining with public HCP Terraform
+  (https://app.terraform.io) is supported in Teleport Community Edition.
+</Admonition>
+
+When running the Teleport Terraform provider on Terraform Cloud, you can use its
+built-in Machine ID support to dynamically authenticate to your Teleport cluster
+without any shared secrets. When run in this configuration, the Terraform
+provider proves its identity to the Teleport Auth Service using Terraform
+Cloud's [Workload Identity][tf-workload-id] tokens.
+
+While following this guide, you'll configure your Teleport cluster to accept
+join requests from Terraform Cloud runs and configure the provider to
+authenticate using the Terraform Cloud join method.
+
+Note that this guide applies to both the public HCP Terraform a.k.a. Terraform
+Cloud, as well as self-hosted Terraform Enterprise. This does **not** apply
+to Terraform or OpenTofu when run on other CI/CD platforms like Spacelift, so
+refer to our generic [CI and Cloud guide](./ci-or-cloud.mdx) to configure the
+provider in these environments.
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+- [Terraform >= (=terraform.version=)+](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+
+  ```code
+  $ terraform version
+  # Terraform v(=terraform.version=)
+  ```
+
+  To import existing Teleport resources as Terraform resources, you must have
+  Terraform version `v1.5.0` or above.
+
+- An account and project on either the public Terraform Cloud SaaS or a
+  Terraform Enterprise instance
+
+- The Teleport Terraform provider, v16.3.0 or later
+
+## Step 1/4: Configure Terraform Cloud joining in Teleport
+
+To start, the Teleport Auth Service needs to be configured to accept join
+requests from Terraform Cloud runs. We'll do this by creating
+[a bot](../../../reference/architecture/machine-id-architecture.mdx#what-is-a-bot)
```

Comment:
Since this link isn't required to complete Step 1, I would move it to a "Further reading" section or delete it (per the style guide: https://github.com/gravitational/docs/blob/main/docs-contributors/style-guide.md#using-links-admonishments-and-details-in-how-to-guides).

Reply:
Fair enough, I'll remove it. The inner workings are mostly irrelevant now that all the Machine ID-y internals are built in to the Terraform Provider.